File: 6 kyu/Counting Duplicates/solution.ex
defmodule DuplicateCount do

  def count(str) do
    str
    |> String.downcase
    |> String.graphemes
    |> Enum.group_by(fn char -> char end)
    |> Enum.count(fn {_, char_group} -> Enum.count(char_group) > 1 end)
  end

end
____________________________
defmodule DuplicateCount do

  def count(str) do
    str 
    |> String.downcase
    |> String.codepoints
    |> Enum.sort
    |> Enum.chunk_by(&(&1))
    |> Enum.filter(&(Enum.count(&1) >1))
    |> Enum.count
  end

end
_______________________
defmodule DuplicateCount do
  def count(str) do
    str
    |> String.downcase()
    |> String.split("", trim: true)
    |> Enum.frequencies()
    |> Enum.filter(fn {_, v} -> v >= 2 end)
    |> Enum.count()
  end
end
_______________
defmodule DuplicateCount do

  def count(str) do
    str
    |> String.downcase()
    |> String.graphemes()
    |> Enum.reduce(%{}, fn ltr, counts -> Map.update(counts, ltr, 1, fn val -> val + 1 end) end) # map like %{"a" => 2, "b" => 1}
    |> Enum.reduce(0, fn {_,count}, dups -> if count > 1, do: dups+1, else: dups end)
  end

end

______________________
defmodule DuplicateCount do

  def count(str) do
    value = str |> String.downcase 
                |> String.graphemes
                |> Enum.reduce(%{}, fn char, acc ->
                     Map.put(acc, char, (acc[char] || 0) + 1)
                   end)
    Map.keys(value) 
    |> Enum.reduce(0, fn char, acc ->
      if value[char] > 1, do: 1 + acc, else: acc
    end)

  end

end

File: 6 kyu/Counting Duplicates/solution.rb
def duplicate_count(text)
  ('a'..'z').count { |c| text.downcase.count(c) > 1 }
end

_____________________
def duplicate_count(text)
  arr = text.downcase.split("")
  arr.uniq.count { |n| arr.count(n) > 1 }
end
____________________
def duplicate_count(text)
  hsh = Hash.new(0)
  text.downcase.chars.each { |c| hsh[c] += 1 }
  hsh.values.count { |k| k > 1 }
end
___________________
def duplicate_count(text)
  text.downcase.chars.group_by(&:to_s).count { |_, v| v.count > 1 }
end
__________________
def duplicate_count(str)
    str.downcase.each_char.find_all { |c| str.downcase.count(c) > 1 }.uniq.size
end
